File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        pairs = {'(': ')', '[': ']', '{': '}'}
        stack = []
        for p in s:
            if p in pairs:
                stack.append(pairs[p])
            elif not stack or stack.pop() != p:
                return False
        return not stack

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
  const stack = [];
  const map = { '(': ')', '[': ']', '{': '}' };
  
  for (let x of s) {
    if (x in map) stack.push(map[x]);
    else if (stack.pop() !== x) return false;
  }
  
  return stack.length === 0;
};